packages/agent/src/agent/http/transforms.ts

```diff
@@ -50,6 +50,14 @@ export class Expiry {
   public toHash(): ArrayBuffer {
     return lebEncode(this._value);
   }
+
+  public toString(): string {
+    return this._value.toString();
+  }
+
+  public toJSON(): string {
+    return this.toString();
+  }
```

Comment:
I had to add this in order to serialize the `Expiry` class in the errors

Reply:
@krpeacock would it be better to have it serialized as an object like we do with the Principal?

```suggestion

  public toString(): string {
    return this._value.toString();
  }

  public toJSON(): { _value: string } {
    return { _value: this.toString() };
  }
```

Reply:
yes, I think it should be an object for consistency. `toJSON` apis conventionally convert to a JSON-serializable format, instead of a serialized string (yes it's weird)

I'd add a matching `fromJSON` method for symmetry

Reply:
I'll make these changes in another PR